### ISSUE TITLE
Documentation on how/why this works

### ISSUE THREAD

**Issue by wparad** [aws/jsii#2281]
There's nothing in the readme that explains how this works, a link or documentation on that would go a long way for understanding. for instance, pure compute may have good 1:1 but how about HTTP calls.

**Comment by SomayaB** [comment]
Hi @wparad, what do you mean by **_this_**? Do you mean jsii as a whole?

**Comment by wparad** [comment]
> Hi @wparad, what do you mean by **_this_**? Do you mean jsii as a whole?

Yes, it sounds like it taking in `.ts` and outputs other languages. Great, but what's the magic, what the high level of how it actually accomplishes the transpilation. Sure in the readme it shows the interface of the other languages but it would really helpful to explain more than just the interface. Take the prime example from [Babel](https://babeljs.io/docs/en/index.html)

```js
// Babel Input: ES2015 arrow function
[1, 2, 3].map((n) => n + 1);

// Babel Output: ES5 equivalent
[1, 2, 3].map(function(n) {
  return n + 1;
});
```
Now this tells me something, I also want jsii to tell me something.

**Comment by MrArnoldPalmer** [comment]
Hey @wparad, I agree we could do with a better high level "how" section in the readme. I'm gonna take a crack at something and I'll add you for review.

**Comment by RomainMuller** [comment]
@MrArnoldPalmer - I reckon this could be a new document in the `Home` section of the ✨shiny✨ new documentation website 😉

**Comment by wparad** [comment]
@MrArnoldPalmer is this coming at some point? Your bot is getting antsy.
